### PR TITLE
[Retention Policy] Stop run once limit is hit

### PR DESCRIPTION
**Pull request by darjoo**
#### Summary 
**Problem**
The RP JQ does not stop running when the limit of records is hit. It continues on to the next table. Whenever the limit is hit, a new JQ will be created.

**Solution**
Set `EndCurrentRun` to true when the limit is hit. That stops the entire RP from proceeding further until the next JQ run.

#### Work Item(s) 
Fixes [AB#497286](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/497286)

### CODE REVIEW

**Review by grobyns — DISMISSED** [review]
no test? it's kinda where this went wrong the first time... :-P

**Review by JesperSchulz — DISMISSED** [review]

**Comment by darjoo** [comment]
> no test? it's kinda where this went wrong the first time... :-P

Yeah, I need to add a new test library and all. Testing first before I push that. :)

File: src/System Application/Test Library/Retention Policy/src/RetentionPolicyTestLibrary.Codeunit.al [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+namespace System.TestLibraries.DataAdministration;
+
+using System.DataAdministration;
+
+codeunit 138709 "Retention Policy Test Library"
+{
+    EventSubscriberInstance = Manual;
+
+    var
+        RecordExceedSubscriberCount: Integer;
```
Comment (grobyns) [inline src/System Application/Test Library/Retention Policy/src/RetentionPolicyTestLibrary.Codeunit.al]: typo? Exceed or Exceeded?

Comment (darjoo) [reply src/System Application/Test Library/Retention Policy/src/RetentionPolicyTestLibrary.Codeunit.al]: Yes, typo :(

**Review by grobyns — APPROVED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**